api/src/main/java/org/apache/iceberg/expressions/Literals.java

```diff
@@ -109,6 +110,26 @@ public T value() {
     public String toString() {
       return String.valueOf(value);
     }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    public boolean equals(Object other) {
```

Comment:
Is this still needed?

Reply:
Yes, it is used for deduplication. Additionally, it might be better to override `equals` to be consistent with the behavior of the comparison.

Reply:
Used for deduplication where?

Reply:
Currently, it has been used in two places.
- In the factory methods in Expressions, when a collection is passed into `in` or `not_in`, they will be converted to Literal and then to a Set.
- In binding, after all literals in Unbound Predicate is converted to the type, it is used for deduplication.